chain-crypto/src/sign.rs

```diff
@@ -120,8 +121,19 @@ impl<A: VerificationAlgorithm, T> Signature<T, A> {
             phantom: PhantomData,
         }
     }
+
+    pub fn safe_coerce<U: SafeSignatureCoerce<T>>(self) -> Signature<U, A> {
```

Comment:
Looks baroque, but safer than coercing `DigestOf` for arbitrary types.
Should we retire that `coerce` method as well?

Reply:
`coerce` is still useful until we converted everything and added a safe version of all the possible conversion we need. At the moment the codebase is slowly converting to typed versions and keeping the raw representation alive, but it's far from being all converted, nor it's a good idea to convert everything in one go.